rom/dev/src/flow/cold_reset/idev_id.rs

```diff
@@ -118,9 +167,20 @@ impl InitDevIdLayer {
     /// * `env` - ROM Environment
     /// * `uds` - Key Vault slot to store the decrypted UDS in
     #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
-    fn decrypt_uds(env: &mut RomEnv, uds: KeyId) -> CaliptraResult<()> {
+    fn decrypt_uds(
+        env: &mut RomEnv,
+        uds_ecdsa: KeyId,
+        uds_mldsa: Option<KeyId>,
+    ) -> CaliptraResult<()> {
         // Engage the Deobfuscation Engine to decrypt the UDS
-        env.doe.decrypt_uds(&DOE_IV, uds)?;
+        env.doe.decrypt_uds(&DOE_IV, uds_ecdsa)?;
+
+        // If Dual-Cert Mode is enabled, recondition the UDS for MLDSA.
+        // [TODO] Use hmac512; set tag usage as seed for mldsa key pair generation.
+        if let Some(uds_mldsa) = uds_mldsa {
+            Crypto::hmac384_mac(env, uds_ecdsa, &b"uds_recon".into(), uds_mldsa)?;
```

Comment:
Is there a reason we use HMAC instead of HKDF for this and the FE below?

Reply:
UDS/FE reconditioning is removed and this no longer applies.